### PR TITLE
core: Remove deprecated load balancer and name resolver APIs

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 1]
```diff
@@ -511,7 +460,7 @@ public void updateOobChannelAddresses(ManagedChannel channel, EquivalentAddressG
      */
     public void updateBalancingState(
         @Nonnull ConnectivityState newState, @Nonnull SubchannelPicker newPicker) {
-      updatePicker(newPicker);
+      throw new UnsupportedOperationException();
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/LoadBalancer.java]: ditto

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 2]
```diff
@@ -128,20 +106,9 @@ public void handleResolvedAddresses(
...
-    for (EquivalentAddressGroup eag : servers) {
-      ResolvedServerInfoGroup.Builder serverInfoGroupBuilder =
-          ResolvedServerInfoGroup.builder(eag.getAttributes());
-      for (SocketAddress addr : eag.getAddresses()) {
-        serverInfoGroupBuilder.add(new ResolvedServerInfo(addr));
-      }
-      serverInfoGroups.add(serverInfoGroupBuilder.build());
-    }
-    handleResolvedAddresses(serverInfoGroups, attributes);
+    throw new UnsupportedOperationException();
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/LoadBalancer.java]: This should be made abstract, since it must be implemented.

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 3]
```diff
@@ -100,26 +98,6 @@
...
-   * @deprecated Implement {@link #handleResolvedAddressGroups} instead.  As it is deprecated, the
-   *             {@link ResolvedServerInfo}s from the passed-in {@link ResolvedServerInfoGroup}s
-   *             lose all their attributes.
-   *
-   * @param servers the resolved server addresses, never empty.
-   * @param attributes extra metadata from naming system.
-   */
-  @Deprecated
-  public void handleResolvedAddresses(
-      List<ResolvedServerInfoGroup> servers, Attributes attributes) {
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/LoadBalancer.java]: ResolvedServerInfoGroup should also be removed.

Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Done

File: core/src/main/java/io/grpc/LoadBalancer.java (hunk 2 above)
Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Done.

File: core/src/main/java/io/grpc/LoadBalancer.java (hunk 1 above)
Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Done

File: core/src/main/java/io/grpc/LoadBalancer.java (hunk 2 above)
Comment (zhangkun83) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Not done yet?

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 4]
```diff
@@ -450,9 +419,8 @@ public boolean equals(Object other) {
      * @throws IllegalArgumentException if {@code subchannel} was not returned from {@link
      *     #createSubchannel}
      */
-    public void updateSubchannelAddresses(Subchannel subchannel, EquivalentAddressGroup addrs) {
-      throw new UnsupportedOperationException();
-    }
+    public abstract void updateSubchannelAddresses(
+        Subchannel subchannel, EquivalentAddressGroup addrs);
```
Comment (zhangkun83) [inline core/src/main/java/io/grpc/LoadBalancer.java]: This change is unrelated to the deletion of the deprecated methods, and it doesn't seem necessary.

Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Sorry, abstracted the wrong method
